src/TranslateTypeDeclarations.re

```diff
@@ -34,16 +34,29 @@ let createCase = ((label, attributes)) =>
   | _ => {label, labelJS: StringLabel(label)}
   };
 
-// Rename record fields.
-// If @genType.as is used, perform renaming conversion.
-// If @bs.as is used (with records-as-objects active), no conversion is required.
+let isJSSafePropertyName = name => {
+  let jsSafeRegex = {|^[A-z][A-z0-9]*$|} |> Str.regexp;
+  Str.string_match(jsSafeRegex, name, 0);
+};
+
+/**
+ * Rename record fields.
+ * If @genType.as is used, perform renaming conversion.
+ * If @bs.as is used (with records-as-objects active), escape and quote if
+ * the identifier contains characters which are invalid as JS property names.
+ */
```

Comment:
I change to doc comments so that it is picked up by intellisense in VS Code

<img width="1405" alt="Screenshot 2020-06-05 at 14 21 53" src="https://user-images.githubusercontent.com/4380655/83880995-ebd70a80-a737-11ea-99f7-f8b9d90d1e2b.png">